bot/exts/evergreen/youtube.py

```diff
@@ -0,0 +1,151 @@
+            title=result.title,
+            url=YOUTUBE_VIDEO_URL.format(id=result.id),
+            post_detail_emoji=Emojis.post_detail,
+            user_emoji=Emojis.user,
+            username=result.username,
+            view_emoji=Emojis.view,
+            view_count=result.video_statistics.view_count,
+            like_emoji=Emojis.like,
+            like_count=result.video_statistics.like_count,
+        )
+
+    async def get_statistics(self, id: str) -> Optional[VideoStatistics]:
+        """Queries API for statistics of one video."""
+        async with self.bot.http_session.get(
+            STATS_API,
+            params={"part": "statistics", "id": id, "key": KEY},
+        ) as response:
+            if response.status != 200:
+                log.error(
+                    f"YouTube statistics response not successful: response code {response.status}"
+                )
+                return None
+
+            statistics = (await response.json())["items"][0]["statistics"]
+
+            return VideoStatistics(
+                view_count=statistics["viewCount"], like_count=statistics["likeCount"]
+            )
+
+    async def search_youtube(self, search: str) -> Optional[List[Video]]:
+        """Queries API for top 5 results matching the search term."""
+        results = []
+        async with self.bot.http_session.get(
+            SEARCH_API,
+            params={"part": "snippet", "q": search, "safeSearch": "strict", "type": "video", "key": KEY},
+        ) as response:
+            if response.status != 200:
+                log.error(
+                    f"YouTube search response not successful: response code {response.status}"
+                )
+                return None
+
+            video_snippet = await response.json()
+
+            for item in video_snippet["items"]:
+                video_statistics = await self.get_statistics(item["id"]["videoId"])
+
+                if video_statistics is None:
+                    log.warning(
+                        "YouTube statistics response not successful, aborting youtube search"
+                    )
+                    return None
+
+                results.append(
+                    Video(
+                        title=escape_markdown(unescape(item["snippet"]["title"])),
+                        username=escape_markdown(
+                            unescape(item["snippet"]["channelTitle"])
+                        ),
+                        id=item["id"]["videoId"],
+                        video_statistics=video_statistics,
+                    )
+                )
+
+        return results
+
+    @commands.command(aliases=["yt"])
+    @commands.cooldown(1, 15, commands.BucketType.user)
+    async def youtube(self, ctx: commands.Context, *, search: str) -> None:
+        """Sends the top 5 results of a query from YouTube with fifteen second cool down per user."""
+        results = await self.search_youtube(search)
+
+        if results:
```

Comment:
1. I would rewrite it as 
```py
if not results:
    ... # sad path
    return
    
# happy path
```
but it's more of a taste question, I guess

Reply:
2. If you don't ever differentiate between `[]` or `None`, why does `search_youtube` return `Optional[List[...]]`? Maybe you should show different errors in each case, or return just `[]` or just `None` in case of error